### PR TITLE
updating prometheus/common to v0.25.0 from v0.10.0

### PR DESCRIPTION
**Pull request by RinkiyaKeDad**
Signed-off-by: RinkiyaKeDad <arshsharma461@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Updates [github.com/prometheus/common](https://github.com/prometheus/common) from v0.10.0 to v0.25.0.

#### Special notes for your reviewer:
Final output from `hack/update-vendor.sh` had this as the last line:
```
+++ [0519 21:03:17] NOTE: don't forget to handle vendor/* files that were added or removed
```
Does something needs to be done in regards to this?

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

[Link](https://kubernetes.slack.com/archives/C20HH14P7/p1621423343002900) to the relevant slack discussion.

/assign @dims

### CODE REVIEW

**Comment by RinkiyaKeDad** [comment]
The `check-dependency-stats` job ran automatically. Yaaaay!

File: go.mod [hunk 1]
```diff
@@ -164,31 +164,40 @@ replace (
...
 	github.com/MakeNowJust/heredoc => github.com/MakeNowJust/heredoc v0.0.0-20170808103936-bb23615498cd
 	github.com/Microsoft/go-winio => github.com/Microsoft/go-winio v0.4.15
 	github.com/Microsoft/hcsshim => github.com/Microsoft/hcsshim v0.8.10-0.20200715222032-5eafd1556990
 	github.com/NYTimes/gziphandler => github.com/NYTimes/gziphandler v1.1.1
 	github.com/PuerkitoBio/purell => github.com/PuerkitoBio/purell v1.1.1
 	github.com/PuerkitoBio/urlesc => github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
+	github.com/Shopify/sarama => github.com/Shopify/sarama v1.19.0
+	github.com/Shopify/toxiproxy => github.com/Shopify/toxiproxy v2.1.4+incompatible
+	github.com/VividCortex/gohistogram => github.com/VividCortex/gohistogram v1.0.0
+	github.com/afex/hystrix-go => github.com/afex/hystrix-go v0.0.0-20180502004556-fa1af6a1f4f5
```
Comment (liggitt) [inline go.mod:177]: what brought all of these in?

File: go.mod [hunk 2]
```diff
@@ -164,31 +164,40 @@ replace (
...
+	github.com/alecthomas/units => github.com/alecthomas/units v0.0.0-20190924025748-f65c72e2690d
 	github.com/andreyvit/diff => github.com/andreyvit/diff v0.0.0-20170406064948-c7f18ee00883
+	github.com/apache/thrift => github.com/apache/thrift v0.13.0
 	github.com/armon/circbuf => github.com/armon/circbuf v0.0.0-20150827004946-bbbad097214e
 	github.com/armon/go-metrics => github.com/armon/go-metrics v0.0.0-20180917152333-f0300d1749da
 	github.com/armon/go-radix => github.com/armon/go-radix v0.0.0-20180808171621-7fddfc383310
+	github.com/aryann/difflib => github.com/aryann/difflib v0.0.0-20170710044230-e206f873d14a
 	github.com/asaskevich/govalidator => github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a
 	github.com/auth0/go-jwt-middleware => github.com/auth0/go-jwt-middleware v0.0.0-20170425171159-5493cabe49f7
+	github.com/aws/aws-lambda-go => github.com/aws/aws-lambda-go v1.13.3
```
Comment (liggitt) [inline go.mod:189]: what brought in new cloud provider libs?

File: go.mod [hunk 3]
```diff
@@ -164,31 +164,40 @@ replace (
...
 	github.com/aws/aws-sdk-go => github.com/aws/aws-sdk-go v1.35.24
+	github.com/aws/aws-sdk-go-v2 => github.com/aws/aws-sdk-go-v2 v0.18.0
```
Comment (liggitt) [inline go.mod:191]: what brought in new cloud provider libs?

File: go.mod [hunk 4]
```diff
@@ -315,30 +336,34 @@ replace (
 	github.com/heketi/heketi => github.com/heketi/heketi v10.2.0+incompatible
 	github.com/heketi/tests => github.com/heketi/tests v0.0.0-20151005000721-f3775cbcefd6
 	github.com/hpcloud/tail => github.com/hpcloud/tail v1.0.0
+	github.com/hudl/fargo => github.com/hudl/fargo v1.3.0
 	github.com/ianlancetaylor/demangle => github.com/ianlancetaylor/demangle v0.0.0-20181102032728-5e5cf60278f6
 	github.com/imdario/mergo => github.com/imdario/mergo v0.3.5
 	github.com/inconshreveable/mousetrap => github.com/inconshreveable/mousetrap v1.0.0
+	github.com/influxdata/influxdb1-client => github.com/influxdata/influxdb1-client v0.0.0-20191209144304-8bf82d3c094d
```
Comment (liggitt) [inline go.mod:343]: what brought in a db client dependency?

**Comment by liggitt** [comment]
This looks to have brought in a startling number of additional dependencies, and is the sort of thing we want to catch/prevent with our dependency checking.

Use `go mod why` or `go mod graph` to determine what is causing these new dependencies. An issue should be opened for the upstream that added them to request they be dropped/isolated so downstream consumers (like us) don't have to be exposed to them.

**Comment by dims** [comment]
@liggitt found the source of the boatload of deps -  https://github.com/prometheus/common/issues/255

**Comment by liggitt** [comment]
> found the source of the boatload of deps - [prometheus/common#255](https://github.com/prometheus/common/issues/255)

that's exceptionally unfortunate, especially since the library doesn't seem concerned about the issue

**Comment by roidelapluie** [comment]
> > found the source of the boatload of deps - [prometheus/common#255](https://github.com/prometheus/common/issues/255)
> 
> that's exceptionally unfortunate, especially since the library doesn't seem concerned about the issue

Can you please be more specific about why you think it is an issue? As explained in the thread, go.sum does not contain the dependencies.

**Comment by dims** [comment]
@RinkiyaKeDad given that promethus/common and go-kit folks will take their own time given comments in  https://github.com/prometheus/common/issues/255 and https://github.com/prometheus/common/pull/302

Let's not waste our time. If it happens we can update then.

**Comment by RinkiyaKeDad** [comment]
Closing till we figure out a way to avoid the extra dependencies which come with github.com/go-kit/kit

**Comment by peterbourgon** [comment]
The indirect dependencies appear in your go.mod, but unless you use them, they aren't brought into the build, and they don't make it into any artifacts. I completely understand the desire to keep dependency graphs small, but there is limited effect on the module, and zero effect on any artifacts.

**Comment by dims** [comment]
@peterbourgon Ack and thanks! we are still trying and it's slowly getting better with every release.

**Pull request closed without merging**